File: test-any/go/write.go
package main

import (
	"os"

	"google.golang.org/protobuf/proto"
	"google.golang.org/protobuf/types/known/anypb"
	"google.golang.org/protobuf/types/known/timestamppb"
)

func write(fname string) {

	p := Person{
		Name:  "John Doe",
		Id:    1,
		Email: "john@d.oe",
	}

	wrappedP, err := anypb.New(&p)
	check(err)

	r := Robot{
		Name: "R2D2",
		Id:   2,
		Features: []*Robot_Feature{
			{
				Name: "Blue",
			},
		},
	}

	wrappedR, err := anypb.New(&r)
	check(err)

	t1 := Task{
		Title:   "Buy Milk",
		DueDate: timestamppb.Now(),
		DoneBy:  wrappedP,
	}

	t2 := Task{
		Title:   "Buy Bread",
		DueDate: timestamppb.Now(),
		DoneBy:  wrappedR,
	}

	f, err := os.Create(fname)
	check(err)

	t1d, err := proto.Marshal(&t1)
	check(err)

	_, err = f.Write(t1d)
	check(err)

	t2d, err := proto.Marshal(&t2)
	check(err)

	_, err = f.Write(t2d)
	check(err)

	check(f.Close())
}

func check(err error) {
	if err != nil {
		panic(err)
	}
}

File: test-any/py/write.py
import schema_pb2

import google.protobuf.any_pb2 as any

person = schema_pb2.Person()
person.name = "John"
person.age = 42

any_person = any.Any()
any_person.Pack(person)

task1 = schema_pb2.Task(done_by=any_person)
task1.title = "task1"
task1.description = "task1 description"

with open("task1.bin", "wb") as f:
    f.write(task1.SerializeToString())
    print("task1.bin written")

robot = schema_pb2.Robot()
robot.name = "R2D2"

any_robot = any.Any()
any_robot.Pack(robot)

task2 = schema_pb2.Task(done_by=any_robot)
task2.title = "task2"
task2.description = "task2 description"

with open("task2.bin", "wb") as f:
    f.write(task2.SerializeToString())
    print("task2.bin written")